ServerCore/Pages/Hints/Responses.cshtml.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.Helpers;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Hints
+{
+    [Authorize(Policy = "IsEventAdminOrEventAuthor")]
+    public class ResponsesModel : EventSpecificPageModel
+    {
+        public ResponsesModel(PuzzleServerContext serverContext, UserManager<IdentityUser> userManager) : base(serverContext, userManager)
+        {
+        }
+
+        public List<HintView> HintViews { get; set; }
+
+        public SortOrder? Sort { get; set; }
+
+        public const SortOrder DefaultSort = SortOrder.PuzzleAscending;
+
+        public async Task<IActionResult> OnGetAsync(SortOrder? sort)
+        {
+            Sort = sort;
+
+            if (EventRole == EventRole.admin)
+            {
+                HintViews = await _context.Hints.Where(h => h.Puzzle.Event == Event)
+                    .Select(h => new HintView { PuzzleId = h.Puzzle.ID, PuzzleName = h.Puzzle.Name, Description = h.Description, Content=h.Content, Cost = h.Cost })
+                    .ToListAsync();
+            }
+            else
+            {
+                // Surely there is a way to get a join to do a bunch of this work, but joins are simply not for me. Someone else can fix later.
```

Comment:
To do this sort of join, you could grab the query from GetPuzzlesForAuthorAndEvent

Reply:
And it might be nice to do so since this change has a little less time pressure.